src/mon/PGMonitor.h

```diff
@@ -32,16 +32,15 @@ using namespace std;
 #include "msg/Messenger.h"
 #include "mon/MonitorDBStore.h"
 
-class MStatfs;
-class MMonCommand;
-class MGetPoolStats;
-
-class TextTable;
 class MPGStats;
+class PGStatService;
 
 class PGMonitor : public PaxosService {
-public:
   PGMap pg_map;
+  std::unique_ptr<PGStatService> pgservice;
+
+  bool do_delete = false;   ///< propose deleting pgmap data
+  bool did_delete = false;  ///< we already deleted pgmap data
```

Comment:
Yeah, I was just trying to be general about it. Simple hacks are simple, though!